src/SignalR/common/Http.Connections/test/MapConnectionHandlerTests.cs

```diff
@@ -68,6 +69,69 @@ public void MapConnectionHandlerFindsAuthAttributesOnDoubleAuthEndPoint()
             Assert.Equal(2, authCount);
         }
 
+        [Fact]
+        public void MapConnectionHandlerEndPointRoutingFindsAttributesOnHub()
+        {
+            var authCount = 0;
+            using (var host = BuildWebHostWithEndPointRouting(routes => routes.MapConnectionHandler<AuthConnectionHandler>("/path", options =>
+            {
+                authCount += options.AuthorizationData.Count;
+            })))
+            {
+                host.Start();
+
+                var dataSource = host.Services.GetRequiredService<EndpointDataSource>();
+                // We register 2 endpoints (/negotiate and /)
+                Assert.Equal(2, dataSource.Endpoints.Count);
+                Assert.NotNull(dataSource.Endpoints[0].Metadata.GetMetadata<IAuthorizeData>());
+                Assert.NotNull(dataSource.Endpoints[1].Metadata.GetMetadata<IAuthorizeData>());
+            }
+
+            Assert.Equal(1, authCount);
+        }
+
+        [Fact]
+        public void MapConnectionHandlerEndPointRoutingAppliesAttributesBeforeConventions()
+        {
+            void ConfigureRoutes(IEndpointRouteBuilder endpoints)
+            {
+                // This "Foo" policy should override the default auth attribute
```

Comment:
Should this be additive instead of override?

Reply:
No.

Reply:
Auth metadata is combined 😆

Reply:
Ah the middleware does it. TIL